### PR TITLE
fix: do not block startup while waiting for kube resource refresh

### PR DESCRIPTION
**Pull request by evanshortiss**
### What does this PR do?

If the current cluster context defined in the user's kube config is not accessible, the startup of Podman Desktop can take 1-2 minutes. This PR fixes the slow initialisation issue mentioned by me in #1633. 

This PR may not be the optimal fix, but I am opening it in case it is considered a valid temporary workaround.

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

#1633 

### How to test this PR?

These steps are specific to my scenario. I assume the same steps could work with Minikube etc.

1. Deploy OpenShift local on a remote Fedora machine.
2. `oc login` to the remote instance form the machine running Podman Deskop (to update `~/.kube/config`)
3. Power down the remote machine.
4. Start Podman Desktop

The Podman Desktop "initialising" screen should be displayed for 1-2 minutes, then log that it failed to fetch API groups before Podman Desktop finishes initialising as shown in this screenshot:

Screenshot 2023-06-15 at 10 57 19 AM

### CODE REVIEW

**Comment by benoitf** [comment]
Thanks @evanshortiss for your PR

**Comment by benoitf** [comment]
If you can amend your commit with a SignedOff

**Review by cdrage — APPROVED** [review]
The code LGTM, it doesn't *seem* like a temporary solution but rather than a permanent one, because the refresh propagates anyways throughout podman desktop / gets put into stores, right @benoitf ?

Tested and works. I purposely used a bad kubeconfig and it no longer got stuck for a minute before loading (although I enjoyed watching the seal gif loading haha)

**Comment by benoitf** [comment]
yes it's fine ,just need to sign the commit 'git commit --amend -s'

File: packages/main/src/plugin/kubernetes-client.ts [hunk 1]
```diff
@@ -168,7 +168,9 @@ export class KubernetesClient {
       // check if path exists
       if (existsSync(userKubeconfigPath)) {
         this.kubeconfigPath = userKubeconfigPath;
-        await this.refresh();
+        this.refresh()
+          .then(() => console.log('Refresh of kube resources on startup complete'))
+          .catch(() => console.log('Refresh of kube resources on startup failed'));
```
Comment (benoitf) [inline packages/main/src/plugin/kubernetes-client.ts]:
```suggestion
          .catch(() => console.error('Refresh of kube resources on startup failed'));
```

File: packages/main/src/plugin/kubernetes-client.ts [hunk 2]
```diff
@@ -168,7 +168,9 @@ export class KubernetesClient {
       // check if path exists
       if (existsSync(userKubeconfigPath)) {
         this.kubeconfigPath = userKubeconfigPath;
-        await this.refresh();
+        this.refresh()
+          .then(() => console.log('Refresh of kube resources on startup complete'))
```
Comment (benoitf) [inline packages/main/src/plugin/kubernetes-client.ts]: Not sure if we need a trace when it completes

Comment (evanshortiss) [reply packages/main/src/plugin/kubernetes-client.ts]: LMK if you need this addressed. I believe the sign-off is all good now 👍

Comment (benoitf) [inline packages/main/src/plugin/kubernetes-client.ts]:
```suggestion
```

**Review by jeffmaury — APPROVED** [review]
LGTM. Thanks @evanshortiss

**Pull request merged**